### PR TITLE
GitHub Workflows security hardening

### PR DESCRIPTION
**Pull request by sashashura**
## Description

This PR adds explicit [permissions section](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions) to workflows. This is a security best practice because by default workflows run with [extended set of permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token) (except from `on: pull_request` [from external forks](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/)). By specifying any permission explicitly all others are set to none. By using the principle of least privilege the damage a compromised workflow can do (because of an [injection](https://securitylab.github.com/research/github-actions-untrusted-input/) or compromised third party tool or action) is restricted.
It is recommended to have [most strict permissions on the top level](https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions) and grant write permissions on [job level](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs) case by case.

## Non-technical explanation

Use least privileges to run workflows.

## Release notes

(*) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

File: .github/workflows/annotate.yml [hunk 1]
```diff
@@ -10,8 +10,13 @@ defaults:
   run:
     shell: bash --noprofile --norc -euo pipefail {0}
 
+permissions: {}
 jobs:
   annotate:
+    permissions:
+      actions: read  #  to list workflow run artifacts
```
Comment (nineinchnick) [inline .github/workflows/annotate.yml:17]: Please out put comments on the previous line

**Review by nineinchnick — APPROVED** [review]

**Comment by nineinchnick** [comment]
@cla-bot check

**Comment by nineinchnick** [comment]
@sashashura can you rebase? There was a failure in trino-main, it's related to fetching artifacts from Maven Central but we made some changes recently to be more resilient against such issues. In a PR like this, we need to have all runs green.

**Comment by nineinchnick** [comment]
@sashashura some CI jobs are failing. It doesn't look related, but I think at least one of these was addressed recently, so if you'd rebase again that would be nice.

To merge this, it needs to be reviewed by a maintainer. You can try asking on Slack to get more attention here.

**Comment by mosabua** [comment]
:wave: @sashashura  - this PR has become inactive. We hope you are still interested in working on it. Please let us know, and we can try to get reviewers to help with that.

We're working on closing out old and inactive PRs, so if you're too busy or this has too many merge conflicts to be worth picking back up, we'll be making another pass to close it out in a few weeks.

**Comment by mosabua** [comment]
I am closing this pool request. If you want to re-base @sashashura please proceed, otherwise if you or @nineinchnick or @wendigo or someone else wants to send a replacement that's fine too

**Pull request closed without merging**